rust/libnewsboat/src/logger.rs

```diff
@@ -215,8 +261,58 @@ impl Logger {
     /// error-logfile will be written.
     ///
     /// Calling this doesn't close already opened logs.
-    pub fn set_loglevel(&mut self, level: Level) {
-        self.loglevel = level;
+    pub fn set_loglevel(&self, level: Level) {
+        self.loglevel.store(level as usize, Ordering::SeqCst);
+    }
+
+    /// Returns current maximum "importance level" of the messages that will be written to the log.
+    ///
+    /// For a more detailed explanation, see `set_loglevel()`.
+    pub fn get_loglevel(&self) -> usize {
+        self.loglevel.load(Ordering::Relaxed)
+    }
+}
+
+static GLOBAL_LOGGER: OnceCell<Logger> = OnceCell::INIT;
+
+/// Returns a global logger instance.
+///
+/// This logger exists for the duration of the program. It's better to set the loglevel and
+/// logfiles as early as possible, so no messages are lost.
+pub fn get_instance() -> &'static Logger {
+    GLOBAL_LOGGER.get().expect("Global logger hasn't been initialized yet")
+}
+
+/// Initializes global logger instance.
+///
+/// The instance is the same as returned by `Logger::new()`.
+///
+/// # Panics
+///
+/// The call panics if the global instance is already initialized.
+pub fn init_global_logger() {
```

Comment:
Could we replace this function by using [`lazy_static!`](https://github.com/rust-lang-nursery/lazy-static.rs)? I think the use case of `lazy_static!` is exactly things like that, isn't it?

Reply:
There was a `lazy_static!` there, but I switched away from it in dde088d6a. But you're right: there is no good reason not to make it lazily-initialized. Thanks!